modules/platforms/dotnet/Apache.Ignite.Tests/BasicAuthenticatorTests.cs

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Tests;
+
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Security;
+
+/// <summary>
+/// Tests for <see cref="BasicAuthenticator"/>.
+/// </summary>
+public class BasicAuthenticatorTests : IgniteTestsBase
+{
+    private const string EnableAuthnJob = "org.apache.ignite.internal.runner.app.PlatformTestNodeRunner$EnableAuthenticationJob";
+
+    private bool _authnEnabled;
+
+    [TearDown]
+    public async Task DisableAuthenticationAfterTest()
+    {
+        await EnableAuthn(false);
+
+        Assert.DoesNotThrowAsync(async () => await Client.Tables.GetTablesAsync());
+    }
+
+    [Test]
+    public async Task TestAuthnOnClientNoAuthnOnServer()
+    {
+        var cfg = new IgniteClientConfiguration(GetConfig())
+        {
+            Authenticator = new BasicAuthenticator
+            {
+                Username = "u",
+                Password = "p"
+            }
+        };
```

Comment:
Why not?
```suggestion
        var cfg = new IgniteClientConfiguration(GetConfig(true));
```

Reply:
Because we pass incorrect credentials intentionally - to make sure that they are ignored when authentication is disabled on the server.